client/src/client/client_future.rs

```diff
@@ -46,7 +47,7 @@ impl<S: Stream<Item = Vec<u8>, Error = io::Error> + 'static> ClientFuture<S> {
         stream: Box<Future<Item = S, Error = io::Error>>,
         stream_handle: Box<ClientStreamHandle<Error = ClientError>>,
         loop_handle: &Handle,
-        signer: Option<Signer>,
+        signer: Option<Arc<Signer>>,
```

Comment:
Signer has to be `Arc` b/c of `OpenSSL`. If we drop OpenSSL entirely this `Arc` can go away. It is mainly due to the fact that `SyncClient` and `SecureSyncClient` now use a lazy connection style for their interior connections. The `Signer` is cloned to each instance of the `Client` as it is used.